core/llm.go

```diff
@@ -73,13 +73,17 @@ type LLM struct {
 	err  error
 
 	// History of messages
-	messages []ModelMessage
+	messages []*ModelMessage
```

Comment:
Context: this was a way-off shot in the dark at the OOMing issue encountered in #10562, but I think it makes sense to keep anyway. This slice gets copied here and there, so might as well avoiding copying the full message history.